azure-devops/global/abstracts/script-e2e-tests.yml

```diff
@@ -0,0 +1,91 @@
+parameters:
+  - name: TAG_NAME
+    type: string
+  - name: PLUGIN_NAME
+    type: string
+  - name: OS_PORT
+    type: string
+  - name: OSD_PORT
+    type: string
+
+steps:
+  - template: ./template-clone-and-move-plugin.yml
+    parameters:
+      TAG_NAME: ${{ parameters.TAG_NAME }}
+      PLUGIN_NAME: ${{ parameters.PLUGIN_NAME }}
+
+  - template: ./replace-env-vars.yml
+    parameters:
+      FILE: '$(Agent.TempDirectory)/OpenSearch-Dashboards/plugins/${{ parameters.PLUGIN_NAME }}/.ci/config/opensearch_dashboards.yml'
+      displayName: 'Replace Environment Variables'
+
+  - script: |
+      idToken=$(curl -s --request POST --header 'Content-Length: 0' \
+          --url "$(ROPC_URL)?username=$(ROPC_USERNAME)&password=$(ROPC_PASSWORD)&grant_type=password&scope=openid%20$(ROPC_CLIENT_ID)%20offline_access&client_id=$(ROPC_CLIENT_ID)&response_type=token%20id_token" | grep -oP '(?<="id_token":")[^"]*')
+      echo "$idToken"
+
+      sed "s/ID_TOKEN/$idToken/" -i $(Agent.TempDirectory)/OpenSearch-Dashboards/plugins/${{ parameters.PLUGIN_NAME }}/public/global/functions/fetch_token.ts
+    name: GetCatalogToken
+    displayName: 'Get API Token'
+
+  - script: |
+      cd $(Agent.TempDirectory)/OpenSearch-Dashboards/plugins/${{ parameters.PLUGIN_NAME }}
+      yarn osd bootstrap &&
+      yarn > $(Agent.TempDirectory)/yarn.log 2>&1
+    displayName: 'Updating Repository and Installing Dependencies'
+
+  - task: Bash@3
+    continueOnError: true
+    inputs:
+      targetType: 'inline'
+      script: |
+        cd $(Agent.TempDirectory)/OpenSearch-Dashboards/plugins/${{ parameters.PLUGIN_NAME }}
+        yarn os snapshot > $(Agent.TempDirectory)/snapshot.log 2>&1 &
+        OS_PID=$!
+
+        while ! grep -q "recovered \[0\] indices into cluster_state" $(Agent.TempDirectory)/snapshot.log; do
+          sleep 1
+        done
+
+        yarn debug > $(Agent.TempDirectory)/debug.log 2>&1 &
+        APP_PID=$!
+
+        while ! grep -q "bundles compiled successfully after" $(Agent.TempDirectory)/debug.log; do
+          sleep 1
+        done
+
+        line=$(grep "http server running at" $(Agent.TempDirectory)/debug.log)
+        OSD_URL=$(echo $line | grep -oP 'http://localhost:${{ parameters.OSD_PORT }}/\w{3}')
+
+        APP_CODE=$(echo $OSD_URL | grep -oP '(?<=http://localhost:${{ parameters.OSD_PORT }}/)\w{3}')
```

Comment:
If this `APP_CODE` irritates you, you can set the parameter rewrite the base path to always be the same or never have it. Look at this configuration here: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/config/opensearch_dashboards.yml#L17